### PR TITLE
Refactoring of project structure

### PR DESCRIPTION
**Pull request by egormkn**
- Added Maven framework support
- Implemented modular structure
- Reviewed module dependencies
- Generated gitignore file

Since now anyone can just clone the project and import the root pom.xml in Intellij IDEA or Eclipse

This pull request is quite big but it had to be done sometime ¯\\\_(ツ)_/¯

### CODE REVIEW

File: pluses-echo-server/src/main/main.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/java" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (Shemplo) [inline pluses-echo-server/src/main/main.iml]: System file of IDEA... drop it

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    <packaging>pom</packaging>
+
+    <name>Pluses Progress Project</name>
+
+    <modules>
+        <module>pluses-message</module>
+        <module>pluses-server</module>
+        <module>pluses-client</module>
+        <module>pluses-echo-server</module>
+    </modules>
```
Comment (Shemplo) [inline pom.xml]: But they are not one project. It's 4 absolutely separate modules that have in common only _Pluses Messages_

File: pluses-echo-server/src/main/main.iml (hunk 1 above)
Comment (egormkn) [reply pluses-echo-server/src/main/main.iml]: Dropped in https://github.com/AlienCoffee/Pluses-Progress-Java-Server/commit/115158e2b6d8f8e79a368e866b9464c313b4ec06

File: pom.xml (hunk 2 above)
Comment (egormkn) [reply pom.xml]: That is how such project structure should look like. Each module can be used on its own, without parent pom.xml

**Pull request merged**